client/js/test/cache_map_test.js

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright 2014 Treode, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+	To run: cd to /client/js/ and run 'mocha'
```

Comment:
I tried this and

```
$ mocha

module.js:340
    throw err;
          ^
Error: Cannot find module 'unit.js'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    ... more ...
```

I think you need to hook mocha into grunt, because Gruntfile is the first place I look to run JS tests.

https://www.npmjs.com/package/grunt-mocha-test